### PR TITLE
Documentation update for Contract and AccountClient

### PR DESCRIPTION
**Pull request by war-in**
### **Please check if the PR fulfills these requirements**
- [ ] The formatter, linter, and tests all run without an error
- [ ] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

## **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Api update

## **What is the new behavior (if this is a feature change)?**

Typing for Contract.deploy updated.
Link to Account.cairo updated.

## **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No

### CODE REVIEW

File: starknet_py/net/account/account_client.py [hunk 1]
```diff
@@ -152,11 +152,11 @@ async def create_account(
         """
         Creates the account using
         `OpenZeppelin Account contract
-        <https://github.com/OpenZeppelin/cairo-contracts/blob/main/contracts/Account.cairo>`_
+        <https://github.com/OpenZeppelin/cairo-contracts/blob/main/src/openzeppelin/account/Account.cairo>`_
```
Comment (cptartur) [inline starknet_py/net/account/account_client.py:155]: Actually we currently use [this contract](https://github.com/starkware-libs/cairo-lang/blob/4e233516f52477ad158bc81a86ec2760471c1b65/src/starkware/starknet/third_party/open_zeppelin/Account.cairo) 
We should probably also change that in other places

**Review by cptartur — CHANGES_REQUESTED** [review]

File: starknet_py/net/account/account_client.py (hunk 1 above)
Comment (war-in) [reply starknet_py/net/account/account_client.py:155]: Are there any other places? I've searched, but I didn't find it

Comment (cptartur) [reply starknet_py/net/account/account_client.py:155]: If that's the case just change it here. New account contract is already referenced in 0.8.0 [migration guide](https://starknetpy.readthedocs.io/en/latest/migration_guide.html).

**Review by cptartur — APPROVED** [review]

**Comment by cptartur** [comment]
Good work! Before merging please change commit title to be more descriptive (they are used as commit messages for merge commits on development). You can see [commit history on development](https://github.com/software-mansion/starknet.py/commits/development) for reference.

**Pull request merged**